docs/annotations.md

```diff
@@ -4,16 +4,77 @@
+    C --> E[Annotations]
+    D --> F[Selection]
+    D --> G[Grouping]
+    E --> H[External tools]
+    E --> I[Documentation]
+```
+
+Both Labels and Annotations are assigned to the following objects managed by Percona Operator for MySQL:
+
+* Custom Resource Definitions
+* Custom Resources
+* Deployments
+* Services
+* StatefulSets
+* PVCs
+* Pods
+* ConfigMaps and Secrets
+
+## When to use labels and annotations
+
+Use **Labels** when:
+
+* The information is used for object selection
+* The data is used for grouping or filtering
+* The information is used by Kubernetes controllers
+* The data is used for operational purposes
+
+Use **Annotations** when:
+
+* The information is for external tools
+* The information is used for debugging
+* The data is used for monitoring configuration
+
+## Labels and annotations used by Percona Operator for MySQL
+
+### Labels
+
+| Name                        | Objects                          | Description                                      | Example values                          |
+|-----------------------------|-----------------------------------|------------------------------------------------|-----------------------------------------|
+| `app.kubernetes.io/name`      | Services, StatefulSets, Deployments, etc. | Specifies the name of the application          | percona-server  |
+| `app.kubernetes.io/instance`  | Services, StatefulSets, Deployments | Identifies a specific instance of the application | cluster1 |
+| `app.kubernetes.io/managed-by`| Services, StatefulSets           | Indicates the controller managing the object    | percona-server-operator |
+| `app.kubernetes.io/component`| Services, StatefulSets           | Specifies the component within the application  | mysql, haproxy, router                  |
+| `app.kubernetes.io/part-of`   | Services, StatefulSets           | Indicates the higher-level application the object belongs to | percona-server                          |
+| `app.kubernetes.io/version`  | CustomResourceDefinition          | Specifies the version of the Percona MySQL Operator. | {{release}} |
+|`percona.com/exposed` | Services | Indicates if the service is exposed externally | true, false |
+
+### Annotations
+
+| Name                                              | Objects  | Description                                   | Example Values              |
+|---------------------------------------------------|----------|-----------------------------------------------|-----------------------------|
+| `service.beta.kubernetes.io/aws-load-balancer-backend-protocol` | Services | Specifies the protocol for AWS load balancers | http, http-test             |
+| `service.beta.kubernetes.io/aws-load-balancer-backend`         | Services | Specifies the backend type for AWS load balancers | test-type                  |
+| `controller-gen.kubebuilder.io/version`     | CustomResourceDefinition          | Indicates the version of the Kubebuilder controller-gen tool used.             |
```

Comment:
there's also last-config-hash and another one related to TLS. @nmarukovich could you please help @nastena1606 ?